### PR TITLE
Change unit tests coverage

### PR DESCRIPTION
**Pull request by mohierf**
As of #587 use a more recent and maintained tests discovery and runner

-----
As of now, we will be able to use the py.test syntax and features for the unit tests.

### CODE REVIEW

File: .travis/unit.sh [hunk 1]
```diff
@@ -6,10 +6,29 @@ cd test
...
-# Run all the unit tests
-nosetests -xv --process-restartworker --processes=1 --process-timeout=300  --with-coverage --cover-package=alignak
+# Run all the unit tests from nose with coverage plugins
+#nosetests -xv --process-restartworker --processes=1 --process-timeout=300  --with-coverage --cover-package=alignak
+
+# Declare a COVERAGE_PROCESS_START environment variable
+# This variable is used to allow coverage tests in the Alignak daemons started processes
+COVERAGE_PROCESS_START='.coveragerc'
+
+# Run all the unit tests from coverage starting nose2 as a module
```
Comment (Seb-Solon) [inline .travis/unit.sh]: Clean the file (duplication / not needed lines)

File: alignak/objects/module.py [hunk 2]
```diff
@@ -77,6 +77,9 @@ class Module(Item):
 
     macros = {}
 
+    def __init__(self, params=None, parsing=True):
```
Comment (Seb-Solon) [inline alignak/objects/module.py]: This seems to be useless

File: test/.coveragerc [hunk 3]
```diff
@@ -1,6 +1,20 @@
 [report]
-omit =
-    */python?.?/*
-    */site-packages/nose/*
+;fail_under = 100
+exclude_lines =
```
Comment (Seb-Solon) [inline test/.coveragerc]: Did you took those examples from somewhere?

File: test/unittest.cfg [hunk 4]
```diff
@@ -0,0 +1,27 @@
+[unittest]
```
Comment (Seb-Solon) [inline test/unittest.cfg]: Short explaination?

File: .travis/unit.sh (hunk 1 above)
Comment (mohierf) [inline .travis/unit.sh]: Yes, of course. I will make some cleanings in all the files...

File: test/.coveragerc (hunk 3 above)
Comment (mohierf) [inline test/.coveragerc]: Yes, and they suit our needs. Those examples are from the Coverage module author. They are what he suggests as most common parameters for coverage

**Comment by mohierf** [comment]
Seems legit to remove this function now

**Pull request merged**